test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs

```diff
@@ -453,6 +461,372 @@ public void TestCommandInvalidArguments(string command, int badCommandIndex)
             }
         }
 
+        [Theory]
+        [InlineData("net5.0")]
+        public async Task AddPkg_WithV2_RelativePath_NoVersionSpecified_Success(string targetFrameworks)
+        {
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var projectName = "projectA";
+                SimpleTestProjectContext projectA = XPlatTestUtils.CreateProject(projectName, pathContext, targetFrameworks);
+                var packageX = "packageX";
+                var packageX_V1 = new PackageIdentity(packageX, new NuGetVersion("1.0.0"));
+                var packageX_V2 = new PackageIdentity(packageX, new NuGetVersion("2.0.0"));
+                var packageXPath = Path.Combine(pathContext.WorkingDirectory, "Custompackages");
+                var sourceRelativePath = RuntimeEnvironmentHelper.IsWindows ? "..\\..\\Custompackages" : "../../Custompackages";
+
+                // Generate Package
+                await SimpleTestPackageUtility.CreateFolderFeedV2Async(
+                    packageXPath, //not using solution source folder
+                    new PackageIdentity[] { packageX_V1, packageX_V2 });
+
+                var projectDirectory = Path.Combine(pathContext.SolutionRoot, projectName);
+                var projectFilePath = Path.Combine(projectDirectory, $"{projectName}.csproj");
+
+                // Act
+                CommandRunnerResult result = _fixture.RunDotnet(projectDirectory, $"add {projectFilePath} package {packageX} -s {sourceRelativePath}", ignoreExitCode: true);
+
+                // Assert
+                result.Success.Should().BeTrue(because: result.AllOutput);
+
+                // Make sure source is replaced in generated dgSpec file.
+                PackageSpec packageSpec = projectA.AssetsFile.PackageSpec;
+                string[] sources = packageSpec.RestoreMetadata.Sources.Select(s => s.Name).ToArray();
+                Assert.Equal(sources.Count(), 1);
+                Assert.True(sources[0].Contains("Custompackages"));
```

Comment:
The value here should be the full path, so you can check it against the full path I think.